### PR TITLE
Tests for changes introduced in ES2015 (Annex E)

### PR DESCRIPTION
**Pull request by anba**
- String case functions iterate over code points
- Has called before Get in Array.p.reverse
- Add test for web-compat Array.p.splice behaviour; Plus missing test for no arguments case
- ToObject no longer applied to this-value in Array.p.toLocaleString
- ToObject no longer applied to this-value in Object.p.toLocaleString
- Add tests for Object.p.propertyIsEnumerable and symbol property keys
- Add tests for Object.p.hasOwnProperty and symbol property keys
- Test property descriptor attributes of message property
- Tests for RegExp constructor checks
- Date constructor when called with date object
- TimeClip never returns negative zero

### CODE REVIEW

File: test/built-ins/Error/message_property.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+---*/
+
+var message = "my-message";
+var error = new Error(message);
+var desc = Object.getOwnPropertyDescriptor(error, "message");
+
+assert.sameValue(desc.value, message, "Property descriptor [[Value]]");
+assert.sameValue(desc.writable, true, "Property descriptor [[Writable]]");
+assert.sameValue(desc.enumerable, false, "Property descriptor [[Enumerable]]");
+assert.sameValue(desc.configurable, true, "Property descriptor [[Configurable]]");
```
Comment (rwaldron) [inline test/built-ins/Error/message_property.js]: These should use the `verify*` property helper functions: 

```
includes:
    - propertyHelper.js
```

```js
verifyWritable(error, "message");
verifyNotEnumerable(error, "message");
verifyConfigurable(error, "message");
```
Example: https://github.com/tc39/test262/blob/master/test/built-ins/Array/from/Array.from-descriptor.js

File: test/built-ins/Error/message_property_native_error.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  var nativeError = nativeErrors[i];
+
+  var message = "my-message";
+  var error = new nativeError(message);
+  var desc = Object.getOwnPropertyDescriptor(error, "message");
+
+  assert.sameValue(desc.value, message, "Property descriptor [[Value]]");
+  assert.sameValue(desc.writable, true, "Property descriptor [[Writable]]");
+  assert.sameValue(desc.enumerable, false, "Property descriptor [[Enumerable]]");
+  assert.sameValue(desc.configurable, true, "Property descriptor [[Configurable]]");
```
Comment (rwaldron) [inline test/built-ins/Error/message_property_native_error.js]: Apply same use of `verify*` property helper functions here.

**Comment by rwaldron** [comment]
Once the descriptor tests are updated, this is good to go

**Comment by anba** [comment]
Updated both tests to use the propertyHelper functions and rebased the PR.

**Comment by rwaldron** [comment]
@anba thank you again!

**Pull request merged**